pages/people/[id]/allocations/add.tsx

```diff
@@ -5,15 +5,31 @@ import AddAllocatedWorker from 'components/AllocatedWorkers/AddAllocatedWorker/A
 import PersonView from 'components/PersonView/PersonView';
 import { useAuth } from 'components/UserContext/UserContext';
 import { isBrowser } from 'utils/ssr';
+import { canUserAllocateWorkerToPerson } from '../../../../lib/permissions';
+import { User } from '../../../../types';
+import { useResident } from '../../../../utils/api/residents';
+import ErrorMessage from '../../../../components/ErrorMessage/ErrorMessage';
+import Spinner from '../../../../components/Spinner/Spinner';
 
 const AddNewAllocationPage = (): React.ReactElement => {
   const { query, replace } = useRouter();
   const personId = Number(query.id as string);
-  const { user } = useAuth();
-  if (isBrowser() && !user?.hasAllocationsPermissions) {
+  const { user } = useAuth() as { user: User };
+  const { data: person, error } = useResident(personId);
+
+  if (error) {
+    return <ErrorMessage />;
+  }
+
+  if (!person) {
+    return <Spinner />;
```

Comment:
What is the Spinner component? Is it the loading circle?

Reply:
Yeah that's right, this one:

<img width="117" alt="Screenshot 2021-06-16 at 15 48 50" src="https://user-images.githubusercontent.com/1573022/122241328-5f1e6b00-ceba-11eb-919e-9be71c332be1.png">